### PR TITLE
chore: Upgrades and syncs Stripe across all packages

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Upgrades Stripe from 9.16 to 14.5 and sync across al packages.
- Fixes some typings with the new versions

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- yarn e2e
- yarn build

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by zomars** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12500** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12500?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12500?utm_source=stack-comment).

File: apps/web/lib/metadata.ts [hunk 1]
```diff
@@ -28,7 +28,6 @@ export const prepareRootMetadata = (recipe: RootMetadataRecipe): Metadata => ({
       {
         rel: "icon-mask",
         url: "/safari-pinned-tab.svg",
-        // @ts-expect-error TODO available in the never Next.js version
```
Comment (zomars) [inline apps/web/lib/metadata.ts:31]: Unused

File: packages/app-store/stripepayment/api/callback.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   const data: StripeData = { ...response, default_currency: "" };
   if (response["stripe_user_id"]) {
     const account = await stripe.accounts.retrieve(response["stripe_user_id"]);
-    data["default_currency"] = account.default_currency;
+    data["default_currency"] = account.default_currency || "";
```
Comment (zomars) [inline packages/app-store/stripepayment/api/callback.ts:41]: Type fix

**Review by zomars — COMMENTED** [review]
Self review done

**Review by CarinaWolli — DISMISSED** [review]

File: packages/app-store/package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "lodash": "^4.17.21",
     "qs-stringify": "^1.2.1",
     "react-i18next": "^12.2.0",
-    "stripe": "^9.16.0"
+    "stripe": "^14.5.0"
```
Comment (keithwillcode) [inline packages/app-store/package.json:29]: Are we sure this isn’t going to cause issues? This is big jump

File: packages/app-store/_utils/stripe.ts [hunk 4]
```diff
@@ -70,5 +70,5 @@ export async function getStripeCustomerId(user: UserType): Promise<string> {
 
 const stripePrivateKey = process.env.STRIPE_PRIVATE_KEY || "";
 export const stripe = new Stripe(stripePrivateKey, {
-  apiVersion: "2020-08-27",
```
Comment (keithwillcode) [inline packages/app-store/_utils/stripe.ts:73]: Are we sure this isn’t going to cause issues? This is big jump

Comment (emrysal) [reply packages/app-store/_utils/stripe.ts:73]: It is, but I inspected this and the stripe typing is fairly good, types will fail if our interfaces don't match 🙏

**Review by emrysal — DISMISSED** [review]
lgtm :)

**Pull request closed without merging**